src/shogun/lib/parameter_observers/ObservedValueTemplated.h

```diff
@@ -0,0 +1,40 @@
+/*
+* Written (W) 2019 Giovanni De Toni
+*/
+
+#ifndef SHOGUN_OBSERVEDVALUETEMPLATED_H
+#define SHOGUN_OBSERVEDVALUETEMPLATED_H
+
+#include <shogun/base/Parameter.h>
+#include <shogun/lib/parameter_observers/ObservedValue.h>
+
+namespace shogun {
+
+	template<class T>
+	class ObservedValueTemplated : public ObservedValue {
+
+	public:
+
+		/**
+		 * Constructor
+		 * @param step step
+		 * @param name the observed value's name
+		 * @param value the observed value
+		 */
+		ObservedValueTemplated(int64_t step, char * name, T value)
+				: ObservedValue(step, name) {
+			SG_ADD(&v, "value", "Value of the observation", ParameterProperties::NONE);
+		}
+
+		~ObservedValueTemplated() {};
+
+		Any get_any() {
+			return make_any(v);
```

Comment:
What about storing the any and returning a reference? Not sure about overhead ...

Reply:
This is more of a convenience method to make some things works. I'm trying to see if I can remove it somehow :/

Reply:
I guess this method will need to stay here for a bit longer. The only place where is needed is here https://github.com/shogun-toolbox/shogun/pull/4552/files#diff-d4ce27ca628f4ba5cacbb66e265af3d0L114, but it is not completely trivial to remove it.